screeps-game-api/src/objects/impls/room_terrain.rs

```diff
@@ -1,15 +1,37 @@
-use {constants::Terrain, objects::RoomTerrain};
+use stdweb::UnsafeTypedArray;
+
+use {constants::{ReturnCode, Terrain}, objects::RoomTerrain, traits::TryInto};
 
 impl RoomTerrain {
     pub fn constructor(room_name: &str) -> Self {
-        js_unwrap!{new Room.Terrain(@{room_name})}
+        js_unwrap!(new Room.Terrain(@{room_name}))
     }
 
     pub fn get(&self, x: u32, y: u32) -> Terrain {
-        js_unwrap!{@{self.as_ref()}.get(@{x}, @{y})}
+        js_unwrap!(@{self.as_ref()}.get(@{x}, @{y}))
     }
 
     pub fn get_raw_buffer(&self) -> Vec<u8> {
-        js_unwrap!(@{self.as_ref()}.getRawBuffer())
+        let mut buffer: Vec<u8> = vec![0; 2500];
+        self.get_raw_buffer_array_ref(&mut buffer).expect("Panic in get_raw_buffer.");
+        buffer
     }
+
+    pub fn get_raw_buffer_array_ref<'a>(&self, buffer: &'a mut Vec<u8>) -> Result<&'a mut Vec<u8>, ReturnCode> {
```

Comment:
Just bikeshedding here: what would you think of making this `get_raw_buffer_to_vec`?

Current name with the `_ref` suffix implies we'll be returning a cheap reference to something rather than making a new one, which is slightly misleading. There's a precedent for `to_xxx` when filling an existing buffer with bytes with functions like https://doc.rust-lang.org/std/io/trait.Read.html#method.read_to_string.

:+1: to having this as a public alternative to `get_raw_buffer` though!